### PR TITLE
Create tutorial section with introduction (#176)

### PR DESCRIPTION
**Pull request by jawoznia**
Part of #176

### CODE REVIEW

File: src/pages/tutorial.mdx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Tutorial
+
+This module is a collection of guides for creating CosmWasm smart contracts. It will lead you step
+by step, and explain relevant topics from the easiest to the trickier ones.
+
+The idea of these tutorials is not only to tell you about smart contracts API but also to show you
+how to do it in a clean and maintainable way. We will show you patterns that CosmWasm creators
```
Comment (uint) [inline src/pages/tutorial.mdx]:
```suggestion
how to write contracts in a clean and maintainable way. We will show you patterns that CosmWasm creators
```

File: src/pages/tutorial.mdx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Tutorial
+
+This module is a collection of guides for creating CosmWasm smart contracts. It will lead you step
+by step, and explain relevant topics from the easiest to the trickier ones.
+
+The idea of these tutorials is not only to tell you about smart contracts API but also to show you
```
Comment (uint) [inline src/pages/tutorial.mdx]:
```suggestion
The point of these tutorials is not only to tell you about smart contracts API but also to show you
```

File: src/pages/tutorial/setup-environment.mdx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+tags: ["tutorial"]
+---
+
+import { Callout } from "nextra/components";
+
+# Environment setup
+
+## Rust installation
+
+To work with CosmWasm smart contract, you will need rust installed on your machine. If you don't
```
Comment (uint) [inline src/pages/tutorial/setup-environment.mdx]:
```suggestion
To work with CosmWasm smart contract, you will need Rust installed on your machine. If you don't
```

File: src/pages/tutorial/setup-environment.mdx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+have one, you can find installation instructions on
```
Comment (uint) [inline src/pages/tutorial/setup-environment.mdx]:
```suggestion
have it, you can find installation instructions on
```

File: src/pages/tutorial/setup-environment.mdx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+[the Rust website](https://www.rust-lang.org/tools/install).
+
+I assume you are working with a stable Rust channel in this book.
```
Comment (uint) [inline src/pages/tutorial/setup-environment.mdx]:
```suggestion
I assume you are working with the stable Rust channel in this book.
```
There's only one! (I think)

File: src/pages/tutorial/setup-environment.mdx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+The output should look like this:
+
+```shell
+Contract checking 1.2.3
+```
+
+## Verifying the installation
+
+To guarantee you are ready to build your smart contracts, you need to make sure you can build
+examples. Checkout the [`cw-plus`](https://github.com/CosmWasm/cw-plus) repository and run the
```
Comment (uint) [inline src/pages/tutorial/setup-environment.mdx]:
```suggestion
examples. Check out the [`cw-plus`](https://github.com/CosmWasm/cw-plus) repository and run the
```

File: src/pages/tutorial/setup-environment.mdx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+testing command in its folder:
+
+```shell copy filename="TERMINAL"
+git clone git@github.com:CosmWasm/cw-plus.git && cd ./cw-plus && cargo test
+```
+
+You should see that everything in the repository gets compiled and all tests pass.
+
+THe [`cw-plus`](https://github.com/CosmWasm/cw-plus) is a great place to find example contracts -
```
Comment (uint) [inline src/pages/tutorial/setup-environment.mdx]:
```suggestion
[`cw-plus`](https://github.com/CosmWasm/cw-plus) is a great place to find example contracts -
```
Proper name

File: src/pages/tutorial/setup-environment.mdx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+look for them in contracts directory. The repository is maintained by CosmWasm creators, so
```
Comment (uint) [inline src/pages/tutorial/setup-environment.mdx]:
```suggestion
look for them in the `contracts` directory. The repository is maintained by CosmWasm creators, so
```

**Review by uint — APPROVED** [review]
Useful and well-written.

I nitpicked a bit - mostly about grammar. None of this is blocking the PR. I recommend using a grammar checker for documentation. There's Grammarly. I have a plugin for emacs that does the job for me.

File: src/pages/tutorial/setup-environment.mdx (hunk 7 above)
Comment (jawoznia) [reply src/pages/tutorial/setup-environment.mdx]: Other than capital `H`, it's intentional. I don't want to start the sentence with the lowercase.

**Pull request merged**